src/modules/cpu/kernel/erase.hpp

```diff
@@ -0,0 +1,322 @@
+                Rpp32u y2 = (Rpp32u) RPPPRANGECHECK(anchorBoxInfo[count].rb.y, y1, roi.xywhROI.roiHeight);
+
+                Rpp32u pixelLocation = (y1 * srcDescPtr->strides.hStride) + (x1 * srcDescPtr->strides.wStride);
+                Rpp32u boxHeight = y2 - y1 + 1;
+                Rpp32u boxWidth = x2 - x1 + 1;
+
+                T *dstPtrTempR, *dstPtrTempG, *dstPtrTempB;
+                dstPtrTempR = dstPtrImage + pixelLocation;
+                dstPtrTempG = dstPtrTempR + dstDescPtr->strides.cStride;
+                dstPtrTempB = dstPtrTempG + dstDescPtr->strides.cStride;
+                Rpp32u countMul3 = count * 3;
+                T userPixelR = colors[countMul3];
+                T userPixelG = colors[countMul3 + 1];
+                T userPixelB = colors[countMul3 + 2];
+                for (int i = 0; i < boxHeight; i++)
+                {
+                    memset(dstPtrTempR, userPixelR, boxWidth * sizeof(T));
+                    memset(dstPtrTempG, userPixelG, boxWidth * sizeof(T));
+                    memset(dstPtrTempB, userPixelB, boxWidth * sizeof(T));
+                    dstPtrTempR += dstDescPtr->strides.hStride;
+                    dstPtrTempG += dstDescPtr->strides.hStride;
+                    dstPtrTempB += dstDescPtr->strides.hStride;
+                }
+            }
+        }
+        // Erase without fused output-layout toggle 1 channel(NCHW -> NCHW)
+        else if ((srcDescPtr->c == 1) && (srcDescPtr->layout == RpptLayout::NCHW) && (dstDescPtr->layout == RpptLayout::NCHW))
+        {
+            memcpy(dstPtrImage, srcPtrImage, dstDescPtr->strides.nStride * sizeof(T));
+
+            for (int count = 0; count < numBoxes; count++)
+            {
+                Rpp32u x1 = (Rpp32u)RPPPRANGECHECK(anchorBoxInfo[count].lt.x, roi.xywhROI.xy.x, roi.xywhROI.roiWidth);
+                Rpp32u y1 = (Rpp32u)RPPPRANGECHECK(anchorBoxInfo[count].lt.y, roi.xywhROI.xy.y, roi.xywhROI.roiHeight);
+                Rpp32u x2 = (Rpp32u)RPPPRANGECHECK(anchorBoxInfo[count].rb.x, x1, roi.xywhROI.roiWidth);
+                Rpp32u y2 = (Rpp32u)RPPPRANGECHECK(anchorBoxInfo[count].rb.y, y1, roi.xywhROI.roiHeight);
+
+                Rpp32u pixelLocation = (y1 * srcDescPtr->strides.hStride) + (x1 * srcDescPtr->strides.wStride);
+                Rpp32u boxHeight = y2 - y1 + 1;
+                Rpp32u boxWidth = x2 - x1 + 1;
+
+                T *dstPtrTemp;
+                dstPtrTemp = dstPtrImage + pixelLocation;
+
+                T userPixel = colors[count];
+                for (int i = 0; i < boxHeight; i++)
+                {
+                    memset(dstPtrTemp, userPixel, boxWidth * sizeof(T));
+                    dstPtrTemp += dstDescPtr->strides.hStride;
+                }
+            }
+        }
+
+        // Erase without fused output-layout toggle 3 channel(NHWC -> NHWC)
+        else
+        {
+            memcpy(dstPtrImage, srcPtrImage, dstDescPtr->strides.nStride * sizeof(T));
```

Comment:
Would suggest use a separate to function to copy only the ROI instead of this copying entire strided buffer

Reply:
Modified this